.github/workflows/ssm-release.yml

```diff
@@ -0,0 +1,249 @@
+          done
+
+      - name: Publish SSM packages to BAH
+        if: github.event.inputs.public == 'true'
+        run: |
+          REGIONS="me-south-1"
+          for region in ${REGIONS}; do
+            echo "Share package to public in ${region}"
+            aws --region ${region} ssm modify-document-permission \
+              --name "${{ env.SSM_PKG_NAME }}" \
+              --permission-type "Share" \
+              --account-ids-to-add "All"
+          done
+
+      - name: Configure AWS Credentials for CPT
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_CPT_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_CPT_SECRET }}
+          aws-region: af-south-1
+
+      - name: Copy SSM package to S3 in CPT
+        run: |
+          aws s3 cp build/packages/ssm s3://${{ env.SSM_S3_BICKET_CPT }}/ssmfile/${{ github.event.inputs.version }} --recursive
+
+      - name: Create SSM packages in CPT
+        run: |
+          REGIONS="af-south-1"
+          for region in ${REGIONS}; do
+            echo "Create package in ${region}"
+            bash tools/ssm/ssm_create.sh ${{ env.SSM_PKG_NAME }} ${{ github.event.inputs.version }} \
+              ${{ env.SSM_S3_BICKET_BAH }}/ssmfile/${{ github.event.inputs.version }} ${region}
+          done
+
+      - name: Publish SSM packages to CPT
+        if: github.event.inputs.public == 'true'
+        run: |
+          REGIONS="af-south-1"
+          for region in ${REGIONS}; do
+            echo "Share package to public in ${region}"
+            aws --region ${region} ssm modify-document-permission \
+              --name "${{ env.SSM_PKG_NAME }}" \
+              --permission-type "Share" \
+              --account-ids-to-add "All"
+          done
+
+      - name: Configure AWS Credentials for MXP
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_MXP_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_MXP_SECRET }}
+          aws-region: eu-south-1
+
+      - name: Copy SSM package to S3 in MXP
+        run: |
+          aws s3 cp build/packages/ssm s3://${{ env.SSM_S3_BICKET_MXP }}/ssmfile/${{ github.event.inputs.version }} --recursive
+
+      - name: Create SSM packages in MXP
+        run: |
+          REGIONS="eu-south-1"
+          for region in ${REGIONS}; do
+            echo "Create package in ${region}"
+            bash tools/ssm/ssm_create.sh ${{ env.SSM_PKG_NAME }} ${{ github.event.inputs.version }} \
+              ${{ env.SSM_S3_BICKET_BAH }}/ssmfile/${{ github.event.inputs.version }} ${region}
+          done
+
+      - name: Publish SSM packages to MXP
+        if: github.event.inputs.public == 'true'
+        run: |
+          REGIONS="eu-south-1"
+          for region in ${REGIONS}; do
+            echo "Share package to public in ${region}"
+            aws --region ${region} ssm modify-document-permission \
```

Comment:
I think we can merge the shared logic into common shell/python scripts, e.g. HKG, MXP etc. have very similar steps

Reply:
Each step has few commands and steps in Github workflow could provide a clear view of progress. If merge all steps into one script, it may lose some observability.